src/PlugWallet/ledger.ts

```diff
@@ -0,0 +1,177 @@
+import Secp256k1KeyIdentity from "../utils/identity/secpk256k1/identity";
+import PlugWallet from "./base";
+import proto from "../utils/proto";
+import { LEDGER_CANISTER_ID } from "../utils/dfx/constants";
+import { validateAccountId } from "../PlugKeyRing/utils";
+import { getAccountId } from "../utils/account";
+
+const { AccountIdentifier, BlockHeight, ICPTs, Memo, Payment, SendRequest, Certification } = proto.ic_ledger.pb.v1;
+
+class PlugWalletLedger extends PlugWallet {
+    protected ledgerIdentity: GenericSignIdentity;
+
+    constructor(args: PlugWalletArgs) {
+      super(args);
+      this.ledgerIdentity = args.identity;
+      this.identity = Secp256k1KeyIdentity.ephimeral();
+      this.agent = createAgent({fetch: this.fetch, defaultIdentity: this.identity});
+    }
+
+    public async sign(payload: BinaryBlob): Promise<BinaryBlob> {
+      return this.ledgerIdentity.sign(payload);
+    }
+
+    public async transferNFT(args: {
+      token: NFTDetails;
+      to: string;
+    }): Promise<boolean> {
+      throw new Error('Transfer NFT not supported on Ledger');
+      return this.executeWithDelegation(super.transferNFT, args);
+    };
+
+    public async burnXTC(args: { to: string; amount: string }) {
+      throw new Error('Burn XTC not supported on Ledger');
+      return this.executeWithDelegation(super.burnXTC, args);
+    };  
+    
+    public async send(args: {
+      to: string;
+      amount: string;
+      canisterId: string;
+      opts?: TokenInterfaces.SendOpts;
+    }): Promise<TokenInterfaces.SendResponse> {
+      if (args.canisterId !== LEDGER_CANISTER_ID) {
+        throw new Error(`Token ${args.canisterId} not supported on Ledger`)
+      }
+      const to = validateAccountId(args.to) ? args.to : getAccountId(Principal.fromText(args.to));
+
+      if(!validateAccountId(to)) {
+        throw new Error('Invalid Account ID')
+      }
+
+      const canisterId = Principal.fromText(LEDGER_CANISTER_ID);
+
+      const agent = createAgent({defaultIdentity: this.ledgerIdentity});
+
+      const amount = args.amount; // 0.1 ICP
+      const memo = args.opts?.memo || BigInt(0);
+      const maxFee = args.opts?.fee || BigInt(10000);
+
+      const toPb = AccountIdentifier.fromObject({ hash: Uint8Array.from(Buffer.from(to, 'hex')) });
+      const amountPb = ICPTs.fromObject({ e8s: amount });
+      const memoPb = Memo.fromObject({ memo: memo.toString(10) });
+      const payment = Payment.fromObject({ receiverGets: amountPb});
+      const maxFeePb = ICPTs.fromObject({ e8s: maxFee.toString(10) });
+
+      const request = SendRequest.fromObject({to: toPb, payment: payment, memo: memoPb, maxFee: maxFeePb});
+    
+      const submitResponse = await agent.call(canisterId, {
+        methodName: 'send_pb',
+        arg: blobFromUint8Array(SendRequest.encode(request).finish()),
+        effectiveCanisterId: canisterId,
+      });
+    
+      if (!submitResponse.response.ok) {
```

Comment:
`response` is never `undefined` ? otherwise you should validate it to avoid `Cannot read properties of undefined` unexpected error